server/src/main/java/org/elasticsearch/inference/Model.java

```diff
@@ -23,6 +23,20 @@ public Model(ModelConfigurations configurations, ModelSecrets secrets) {
         this.secrets = Objects.requireNonNull(secrets);
     }
 
+    public Model(Model model, TaskSettings taskSettings) {
```

Comment:
These just make it easier to construct the `Model` when these various settings change but the secrets do not.